### ISSUE TITLE
where are the genes going

### ISSUE THREAD

**Issue by jjc2718** [greenelab/mpmp#44]
As more data types are added, we end up with fewer genes (see https://github.com/greenelab/mpmp-manuscript/pull/8). I thought this was because of the inner CV loop ending up with folds with no positive samples, but it looks like sklearn GridSearchCV uses label-stratified CV by default (see https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html).

We should investigate why certain genes are non-deterministically (?) dropping out of certain data types, and think about if there's any way to preserve some of them across analyses.

**Comment by jjc2718** [comment]
Examples of genes that are present in some, but not all, data types:

* GNAQ
* JAK3
* CDC73
* MPL

Looks like the variability is mainly in the shuffled files (e.g. some data type/seed combinations drop out for shuffled files, but signal files seem to be relatively consistent). Maybe we can fix the shuffling order for each random seed?

**Comment by jjc2718** [comment]
It also looks like the cancer type filtering is based on the set of all samples, not the samples that are actually included in the experiment. We should probably be filtering gene/cancer type pairs independently for each sample set.

**Comment by jjc2718** [comment]
Another inconsistency I noticed is that we're doing hyper-mutated sample filtering after we decide which diseases (and therefore genes) to include. This means that some genes may end up with considerably fewer than 15 mutated samples after filtering for hyper-mutators.
